server/services/property.service.js

```diff
@@ -53,12 +53,39 @@ export const updateProperty = async (updatedProperty) => {
     throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
   });
 
+  let { units, availableUnits } = property;
+
+  if (updatedProperty.units) {
+    const initialUnits = property.units;
+    const updatedunits = updatedProperty.units;
+    const unitDifference = updatedunits - initialUnits;
+
+    if (updatedunits > initialUnits) {
+      units = updatedunits;
+      availableUnits += unitDifference;
+    }
+
+    if (updatedunits < initialUnits) {
+      units = updatedunits;
+      availableUnits += unitDifference;
+
+      if (availableUnits < 0) {
+        throw new ErrorHandler(
+          httpStatus.PRECONDITION_FAILED,
+          'Available property units cannot be less than zero',
+        );
+      }
+    }
+  }
+
+  const propertyToUpdate = { ...updatedProperty, units, availableUnits };
```

Comment:
Units is already defined in `updatedProperty`.

```suggestion
  const propertyToUpdate = { ...updatedProperty, availableUnits };
```